src/main/capi/appender-c.cpp

```diff
@@ -208,6 +208,25 @@ duckdb_state duckdb_appender_close(duckdb_appender appender) {
 	return duckdb_appender_run_function(appender, [&](Appender &appender) { appender.Close(); });
 }
 
+duckdb_state duckdb_appender_get_column_types(duckdb_appender appender, DUCKDB_TYPE **out_types) {
```

Comment:
Shouldnt we also check if `out_types` is not null?
Since we dereference it unchecked later on